client/my-sites/theme/main.jsx

```diff
@@ -672,6 +688,52 @@ class ThemeSheet extends Component {
 		page( localizeThemesPath( backPath, locale, ! isLoggedIn ) );
 	};
 
+	getBannerUpsellTitle = () => {
+		const {
+			isBundledSoftwareSet,
+			isExternallyManagedTheme,
+			translate,
+			isPurchased,
+			isSiteEligibleForManagedExternalThemes,
+		} = this.props;
+
+		if ( isBundledSoftwareSet && ! isExternallyManagedTheme ) {
+			return translate( 'Access this WooCommerce theme with a Business plan!' );
+		} else if ( isExternallyManagedTheme ) {
+			if ( ! isPurchased && ! isSiteEligibleForManagedExternalThemes ) {
+				return translate( 'Upgrade to a Business Plan and subscribe to this theme!' );
+			}
+			return translate( 'Subscribe to this Premium Theme!' );
```

Comment:
Minor nitpick: It feels weird to capitalize "Premium Theme" as a proper noun here. I don't think we've done that elsewhere.